--- a/src/zopt.c
+++ b/src/zopt.c
@@ -23,7 +23,7 @@
 
 #include <getopt.h>
 
-#include "/home/qwerty/Nextcloud/Arbeit/2023/IPv6_Neighbors/5_Linux/aim_zmap_reqnr_single/src/zopt.h"
+#include "zopt.h"
 
 const char *gengetopt_args_info_purpose = "A fast Internet-wide scanner.";
 
--- a/src/topt.c
+++ b/src/topt.c
@@ -23,7 +23,7 @@
 
 #include <getopt.h>
 
-#include "/home/qwerty/Nextcloud/Arbeit/2023/IPv6_Neighbors/5_Linux/aim_zmap_reqnr_single/src/topt.h"
+#include "topt.h"
 
 const char *gengetopt_args_info_purpose = "A buffering output splitter";
 
--- a/src/zbopt.c
+++ b/src/zbopt.c
@@ -23,7 +23,7 @@
 
 #include <getopt.h>
 
-#include "/home/qwerty/Nextcloud/Arbeit/2023/IPv6_Neighbors/5_Linux/aim_zmap_reqnr_single/src/zbopt.h"
+#include "zbopt.h"
 
 const char *gengetopt_args_info_purpose = "A tool for limiting and deduplicating a list of IP addresses";
 
--- a/src/zitopt.c
+++ b/src/zitopt.c
@@ -23,7 +23,7 @@
 
 #include <getopt.h>
 
-#include "/home/qwerty/Nextcloud/Arbeit/2023/IPv6_Neighbors/5_Linux/aim_zmap_reqnr_single/src/zitopt.h"
+#include "zitopt.h"
 
 const char *gengetopt_args_info_purpose = "A tool for iterating over the IPv4 space";
 
--- a/src/ztopt.c
+++ b/src/ztopt.c
@@ -23,7 +23,7 @@
 
 #include <getopt.h>
 
-#include "/home/qwerty/Nextcloud/Arbeit/2023/IPv6_Neighbors/5_Linux/aim_zmap_reqnr_single/src/ztopt.h"
+#include "ztopt.h"
 
 const char *gengetopt_args_info_purpose = "A tests harness tool for zmap";
 
