--- a/src/probe_modules/module_icmp6_echoscan_time.c
+++ b/src/probe_modules/module_icmp6_echoscan_time.c
@@ -130,8 +130,8 @@
 		return 0;
 	}
 
-    // IPv6 header is fixed length at 40 bytes + ICMPv6 header + 16 bytes of ICMPv6 data
-	if ( ( sizeof(struct ip6_hdr) + sizeof(struct icmp6_hdr) + 4 * sizeof(uint32_t)) > len) {
+    // IPv6 header is fixed length at 40 bytes + ICMPv6 header + 20 bytes of ICMPv6 data
+	if ( ( sizeof(struct ip6_hdr) + sizeof(struct icmp6_hdr) + 5 * sizeof(uint32_t)) > len) {
 		// buffer not large enough to contain expected icmp header
 		return 0;
 	}
@@ -144,7 +144,7 @@
 	if (icmp6_h->icmp6_type == ICMP6_TIME_EXCEEDED || icmp6_h->icmp6_type == ICMP6_DST_UNREACH
         || icmp6_h->icmp6_type == ICMP6_PACKET_TOO_BIG || icmp6_h->icmp6_type == ICMP6_PARAM_PROB) {
 
-        // IP6 + ICMP6 headers + inner headers + 16 byte payload (validation)
-        if (2*sizeof(struct ip6_hdr) + 2*sizeof(struct icmp6_hdr) + 4*sizeof(uint32_t) > len) {
+        // IP6 + ICMP6 headers + inner headers + 20 byte payload (validation and request number)
+        if (2*sizeof(struct ip6_hdr) + 2*sizeof(struct icmp6_hdr) + 5*sizeof(uint32_t) > len) {
 			return 0;
 		}
@@ -185,5 +185,6 @@
 		fs_add_string(fs, "orig-dest-ip", make_ipv6_str(&(ip6_hdr->ip6_src)), 1);
 		fs_add_uint64(fs, "sent_timestamp_ts", (uint64_t)icmp6_hdr->icmp6_data32[3]);
 		fs_add_uint64(fs, "sent_timestamp_us", (uint64_t)icmp6_hdr->icmp6_data32[4]);
+		fs_add_uint64(fs, "nrsent", (uint64_t)icmp6_hdr->icmp6_data32[5]);
 		fs_add_string(fs, "classification", (char*) "echoreply", 0);
 		fs_add_uint64(fs, "success", 1);
